### PR TITLE
removed the quarantine VLAN

### PR DESCRIPTION
**Pull request by mereacre**
This PR adds the functionality to setup an IP for the subnet interface without creating it.

Other stuff.
 - Removed the quarantine VLAN.
 - various cleanup

### CODE REVIEW

File: src/utils/uci_wrt.c [hunk 1]
```diff
@@ -730,62 +762,61 @@ int uwrt_add_dhcp_pool(struct uctx *context, char *ifname, int vlanid,
...
 
-  sprintf(property, "dhcp.%s%d.dhcpv4=server", ifname, vlanid);
+  sprintf(property, "dhcp.%s.dhcpv4=server", ifname);
   if (uwrt_set_property(context->uctx, property) < 0) {
     log_trace("uwrt_set_property fail for %s", property);
     return -1;
   }
 
-  sprintf(property, "dhcp.%s%d.instance=edgesec", ifname, vlanid);
+  sprintf(property, "dhcp.%s.instance=edgesec", ifname);
```
Comment (aloisklink) [inline src/utils/uci_wrt.c:789]:
```suggestion
  snprintf(property, ARRAY_SIZE(property), "dhcp.%s.instance=edgesec", ifname);
```

File: src/utils/uci_wrt.c [hunk 2]
```diff
@@ -730,62 +762,61 @@ int uwrt_add_dhcp_pool(struct uctx *context, char *ifname, int vlanid,
...
   if (uwrt_set_property(context->uctx, property) < 0) {
     log_trace("uwrt_set_property fail for %s", property);
     return -1;
   }
 
-  sprintf(property, "dhcp.%s%d.ignore=0", ifname, vlanid);
+  sprintf(property, "dhcp.%s.ignore=0", ifname);
```
Comment (aloisklink) [inline src/utils/uci_wrt.c:795]:
```suggestion
  snprintf(property, ARRAY_SIZE(property), "dhcp.%s.ignore=0", ifname);
```

File: src/utils/uci_wrt.c [hunk 3]
```diff
@@ -730,62 +762,61 @@ int uwrt_add_dhcp_pool(struct uctx *context, char *ifname, int vlanid,
...
   if (uwrt_set_property(context->uctx, property) < 0) {
     log_trace("uwrt_set_property fail for %s", property);
     return -1;
   }
 
-  sprintf(property, "dhcp.%s%d.force=1", ifname, vlanid);
+  sprintf(property, "dhcp.%s.force=1", ifname);
```
Comment (aloisklink) [inline src/utils/uci_wrt.c:801]:
```suggestion
  snprintf(property, ARRAY_SIZE(property), "dhcp.%s.force=1", ifname);
```

File: src/utils/uci_wrt.c [hunk 4]
```diff
@@ -730,62 +762,61 @@ int uwrt_add_dhcp_pool(struct uctx *context, char *ifname, int vlanid,
...
   if (uwrt_set_property(context->uctx, property) < 0) {
     log_trace("uwrt_set_property fail for %s", property);
     return -1;
   }
 
-  sprintf(property, "dhcp.%s%d.start=%d", ifname, vlanid, start);
+  sprintf(property, "dhcp.%s.start=%d", ifname, start);
```
Comment (aloisklink) [inline src/utils/uci_wrt.c:807]:
```suggestion
  snprintf(property, ARRAY_SIZE(property), "dhcp.%s.start=%d", ifname, start);
```

File: src/utils/uci_wrt.c [hunk 5]
```diff
@@ -730,62 +762,61 @@ int uwrt_add_dhcp_pool(struct uctx *context, char *ifname, int vlanid,
...
   if (uwrt_set_property(context->uctx, property) < 0) {
     log_trace("uwrt_set_property fail for %s", property);
     return -1;
   }
 
-  sprintf(property, "dhcp.%s%d.limit=%d", ifname, vlanid, limit);
+  sprintf(property, "dhcp.%s.limit=%d", ifname, limit);
```
Comment (aloisklink) [inline src/utils/uci_wrt.c:813]:
```suggestion
  snprintf(property, ARRAY_SIZE(property), "dhcp.%s.limit=%d", ifname, limit);
```

File: src/utils/uci_wrt.c [hunk 6]
```diff
@@ -730,62 +762,61 @@ int uwrt_add_dhcp_pool(struct uctx *context, char *ifname, int vlanid,
...
   if (uwrt_set_property(context->uctx, property) < 0) {
     log_trace("uwrt_set_property fail for %s", property);
     return -1;
   }
 
-  sprintf(property, "dhcp.%s%d.leasetime=%s", ifname, vlanid, lease_time);
+  sprintf(property, "dhcp.%s.leasetime=%s", ifname, lease_time);
```
Comment (aloisklink) [inline src/utils/uci_wrt.c:819]:
```suggestion
  snprintf(property, ARRAY_SIZE(property), "dhcp.%s.leasetime=%s", ifname, lease_time);
```

File: tests/dhcp/test_dnsmasq.c [hunk 7]
```diff
@@ -36,6 +40,17 @@ static char *test_dhcp_conf_content =
...
+    "server=8.8.4.4\n"
+    "server=8.8.8.8\n"
+    "dhcp-leasefile=/tmp/test_dnsmasq.leases\n"
+    "dhcp-script=/tmp/dnsmasq_exec-test.sh\n"
+    "dhcp-range=eth_if0,10.0.0.2,10.0.0.254,255.255.255.0,24h\n"
+    "dhcp-range=eth_if1,10.0.1.2,10.0.1.254,255.255.255.0,24h\n"
+    "dhcp-range=eth_if2,10.0.2.2,10.0.2.254,255.255.255.0,24h\n"
+    "dhcp-range=eth_if3,10.0.3.2,10.0.3.254,255.255.255.0,24h\n";
+
 // why aren't we using amazing C++11 which has the R"(...) string literal??? 😭
```
Comment (aloisklink) [inline tests/dhcp/test_dnsmasq.c:54]: I was literally just about to comment this same thing :sob:

File: tests/dhcp/test_dnsmasq.c [hunk 8]
```diff
@@ -160,9 +172,9 @@ static void test_generate_dnsmasq_conf(void **state) {
   strcpy(dconf.dhcp_conf_path, test_dhcp_conf_path);
   strcpy(dconf.dhcp_script_path, test_dhcp_script_path);
   strcpy(dconf.dhcp_leasefile_path, test_dhcp_leasefile_path);
-  const size_t WIFI_INTERFACE_STR_LEN =
-      sizeof(dconf.wifi_interface) / sizeof(dconf.wifi_interface[0]);
-  strncpy(dconf.wifi_interface, interface, WIFI_INTERFACE_STR_LEN);
+  const size_t WIFI_INTERFACE_STR_LEN = STRLEN(dconf.wifi_interface);
```
Comment (aloisklink) [inline tests/dhcp/test_dnsmasq.c:175]: Ahhhh, I forgot edgesec uses `STRLEN` instead of `ARRAY_LEN`, my bad!

**Review by aloisklink — APPROVED** [review]
Looks good, most of my comments are pretty minor.

Main point is, what is with the `Testing` directory? I'm guessing it was accidentally commited?

Ah, side-note, there are quite a lot of comments I made, so maybe it's easier to view them in the https://github.com/nqminds/edgesec/pull/198/files tab.

File: Testing/Temporary/LastTest.log [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Start testing: Jul 11 15:03 BST
+----------------------------------------------------------
+End testing: Jul 11 15:03 BST
```
[the remark this answers is not in the document]
Comment (mereacre) [reply Testing/Temporary/LastTest.log:3]: Weird. I must have committed by mistake or I was hacked :)

File: src/dhcp/dnsmasq.c [hunk 10]
```diff
@@ -63,11 +63,39 @@
...
+  sprintf(ifname, "%s%d", dconf->bridge_prefix, vlanid);
+#else
+  if (strlen(dconf->wifi_interface)) {
+    if (vlanid) {
+      sprintf(ifname, "%s.%d", dconf->wifi_interface, vlanid);
+    } else {
+      sprintf(ifname, "%s", dconf->wifi_interface);
+    }
+  } else {
+    sprintf(ifname, "%s%d", dconf->interface_prefix, vlanid);
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/dhcp/dnsmasq.c:88]: Good catch.

File: tests/dhcp/test_dnsmasq.c (hunk 7 above)
Comment (mereacre) [reply tests/dhcp/test_dnsmasq.c:54]: Rust :)

File: Testing/Temporary/LastTest.log (hunk 9 above)
[the remark this answers is not in the document]
Comment (aloisklink) [reply Testing/Temporary/LastTest.log:3]: Hahahaha, your directory names with the capital letters also looks very Microsoft® Windows®-y :wink: 

(Microsoft®, Windows® are either registered trademarks or trademarks of Microsoft Corporation in the United States and/or other countries and are only used in this context for parody purposes under [17 U.S.C.](https://en.wikipedia.org/wiki/Title_17_of_the_United_States_Code) [§ 107](https://www.law.cornell.edu/uscode/text/17/107))

File: src/runctl.c [hunk 11]
```diff
@@ -185,6 +185,48 @@ int create_subnet_interfaces(struct iface_context *context,
...
 
+int set_subnet_ips(struct iface_context *context, UT_array *ifinfo_array) {
+  config_ifinfo_t *p = NULL;
+
+  if (ifinfo_array == NULL) {
+    log_error("ifinfo_array param is NULL");
+    return -1;
+  }
+
+  while ((p = (config_ifinfo_t *)utarray_next(ifinfo_array, p)) != NULL) {
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/runctl.c:196]: I will keep the while loop definition, otherwise I have to change in all the places.

File: src/utils/uci_wrt.c (hunk 4 above)
Comment (mereacre) [reply src/utils/uci_wrt.c:807]: `ARRAY_SIZE` - where is it defined?

**Comment by aloisklink** [comment]
Btw, before merging, make sure you click "Load more" on the hidden conversations in case anything is valid there.

image

File: src/utils/uci_wrt.c (hunk 4 above)
Comment (aloisklink) [reply src/utils/uci_wrt.c:807]: I found it in https://github.com/nqminds/edgesec/blob/302f4fd0e9c445037fcceca8aeaea16acef9a858/lib/libnetlink/include/utils.h#L267, but I just remember that in this repo, you normally use `STRLEN` (although maybe it's worth renaming it to ARRAY_LENGTH) https://github.com/nqminds/edgesec/blob/c9bd32a1868992cfe1ff87b013e8765f96d6f16a/src/utils/os.h#L45

File: src/supervisor/sqlite_macconn_writer.h [hunk 12]
```diff
@@ -24,7 +24,7 @@
 #define MACCONN_TABLE_NAME "instance"
 
 #define MACCONN_CREATE_TABLE                                                   \
-  "CREATE TABLE " MACCONN_TABLE_NAME                                           \
+  "CREATE TABLE IF NOT EXISTS " MACCONN_TABLE_NAME                             \
   " (id TEXT NOT NULL, mac TEXT NOT NULL, status INTEGER, vlanid INTEGER, "    \
   "primaryip TEXT, secondaryip TEXT, nat INTEGER, allow INTEGER, label TEXT, " \
   "timestamp INTEGER, pass TEXT, PRIMARY KEY (id, mac));"
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/supervisor/sqlite_macconn_writer.h:30]: Maybe we could it as a issue/PR.

File: src/utils/iface.h [hunk 13]
```diff
@@ -71,6 +71,17 @@ char *iface_get_vlan(char *if_buf);
  */
 UT_array *iface_get(char *ifname);
 
+/**
+ * @brief Get the IP4 addresses for a given interface
+ *
+ * @param context The interface context
+ * @param[in] brname The bridge name
+ * @param[in] ifname The interface name
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/utils/iface.h:79]: Eventually I will change everything to [in], [out].

File: src/runctl.c [hunk 14]
```diff
@@ -185,6 +185,48 @@ int create_subnet_interfaces(struct iface_context *context,
...
+    log_debug("Setting IP for brname=%s ifname=%s ip_addr=%s brd_addr=%s "
+              "subnet_mask=%s",
+              p->brname, p->ifname, p->ip_addr, p->brd_addr, p->subnet_mask);
+    UT_array *iparr = iface_get_ip4(context, p->brname, p->ifname);
+    char **ip = NULL;
+    int found = 0;
+    while ((ip = (char **)utarray_next(iparr, ip)) != NULL) {
+      if (strcmp(*ip, p->ip_addr) == 0) {
+        found = 1;
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/runctl.c:205]: I will keep the while and the change the int to bool.

File: src/utils/uci_wrt.c (hunk 4 above)
Comment (mereacre) [reply src/utils/uci_wrt.c:807]: I changed all definitions.

File: src/utils/uci_wrt.c [hunk 15]
```diff
@@ -477,6 +478,45 @@ UT_array *uwrt_get_interfaces(struct uctx *context, char *ifname) {
...
+    return -1;
+  }
+
+  sprintf(property, "network.%s.ipaddr=%s", ifname, ip_addr);
+  if (uwrt_set_property(context->uctx, property) < 0) {
+    log_trace("uwrt_set_property fail for %s", property);
+    return -1;
+  }
+
+  sprintf(property, "network.%s.netmask=%s", ifname, netmask);
```
[the remark this answers is not in the document]
Comment (mereacre) [reply src/utils/uci_wrt.c:511]: Thanks.

**Pull request merged**